engine/class_modules/monk/sc_monk.cpp

```diff
@@ -4202,6 +4337,43 @@ struct flurry_of_xuen_t : public monk_spell_t
 // Invoke Niuzao, the Black Ox
 // ==========================================================================
 
+struct strength_of_the_black_ox_t : public monk_spell_t
+{
+  strength_of_the_black_ox_t( monk_t *p )
+    : monk_spell_t( p, "strength_of_the_black_ox", p->talent.conduit_of_the_celestials.strength_of_the_black_ox_damage )
+  {
+    background = true;
+  }
+
+  double composite_aoe_multiplier( const action_state_t *state ) const override
+  {
+    double cam = monk_spell_t::composite_aoe_multiplier( state );
+
+    if ( p()->buff.unity_within->check() )
+      cam *= 1 + p()->talent.conduit_of_the_celestials.unity_within_dmg_mult->effectN( 1 ).percent();
+
+    return cam;
+  }
+
+  void execute() override
+  {
+    monk_spell_t::execute();
+  }
+};
+
+struct strength_of_the_black_ox_absorb_t : public monk_absorb_t
+{
+  strength_of_the_black_ox_absorb_t( monk_t *p )
+    : monk_absorb_t( p, "strength_of_the_black_ox_absorb",
+                     p->talent.conduit_of_the_celestials.strength_of_the_black_ox_absorb )
+  {
+    background = true;
+    aoe         = data().effectN( 3 ).base_value();
+    base_dd_min = p->resources.max[ RESOURCE_HEALTH ] * data().effectN( 2 ).percent();
```

Comment:
`p->max_health() * data().effectN( 2 ).percent()` is a more clear way to express this.